flang/lib/Optimizer/CodeGen/DescriptorModel.h

```diff
@@ -89,10 +87,9 @@ TypeBuilderFunc getModel<Fortran::ISO::CFI_type_t>() {
   };
 }
 template <>
-TypeBuilderFunc getModel<Fortran::ISO::CFI_index_t>() {
+TypeBuilderFunc getModel<long>() {
```

Comment:
CFI_index_t should always be 64 bits.  "long" may not be 64 bits in all environments.  If you want a fixed-size type, use std::int64_t.

Reply:
`Fortran::ISO::CFI_index_t` was conflicting with the template specialization for `long long` on the Windows buildbot.
If we remove it we are then missing a template specialization for `long`. It seems that on some Linux platform `ptrdiff_t` is 32 bits.